core/src/main/java/org/elasticsearch/search/aggregations/ParsedAggregation.java

```diff
@@ -36,21 +35,25 @@
 
     //TODO move CommonFields out of InternalAggregation
     protected static void declareCommonFields(ObjectParser<? extends ParsedAggregation, Void> objectParser) {
-        objectParser.declareObject((parsedAgg, metadata) -> parsedAgg.metadata.putAll(metadata),
+        objectParser.declareObject((parsedAgg, metadata) -> parsedAgg.metadata = metadata,
```

Comment:
I pushed a very similar change to our branch. There was a discrepancy between empty and null handling in this class compared to InternalAggregation, I aligned it to InternalAggregation. this should not be needed anymore here.